booster_nrpe/booster_nrpe.py

```diff
@@ -453,69 +433,58 @@ def manage_finished_checks(self):
         to_del = []
 
         # First look for checks in timeout
-        for c in self.checks:
-            if c.status == 'launched':
-                c.con.look_for_timeout()
-
-        # We check if all new things in connections
-        # WARNING : using poll2 instead of poll (poll1 is with select
-        # call that is limited to 1024 connexions, poll2 is ... poll).
-        asyncore.poll2(timeout=1)
+        for check in self.checks:
+            if check.status == 'launched':
+                check.con.look_for_timeout()
 
         # Now we look for finished checks
-        for c in self.checks:
+        for check in self.checks:
             # First manage check in error, bad formed
-            if c.status == 'done':
-                to_del.append(c)
-                try:
-                    self.returns_queue.put(c)
-                except IOError, exp:
-                    logger.error("[NRPEPoller] Exiting: %s" %  exp)
-                    sys.exit(2)
-                continue
+            if check.status == 'done':
+                to_del.append(check)
+                self.returns_queue.put(check)
```

Comment:
As a sub process the father can crash. We also saw in the past closed fd or IOErrors without the father process to be really dead (especially on very very busy servers, cannot reproduce in a clean way :( ). So if we can catch and clean exit (log it in fact) it's cool. Then the father (if still alive ^^ ) can relaunch this worker, and so recreate the pipe/queue ^^